src/actions/publish.js

```diff
@@ -96,24 +96,22 @@ export function attemptToPublish(): ThunkAction<Promise<boolean>> {
 
       // Get the current generation of this request. It can be aborted midway through.
       // This way we can check inside this async function if we need to bail out early.
-      const uploadGeneration = getUploadGeneration(getState());
+      const uploadGeneration = getUploadGeneration(prePublishedState);
 
       dispatch(uploadCompressionStarted());
-      const gzipData: Uint8Array = await getSanitizedProfileData(getState());
+      const gzipData: Uint8Array = await getSanitizedProfileData(
+        prePublishedState
+      );
 
       // The previous line was async, check to make sure that this request is still valid.
+      // The upload could have been aborted while we were compressing the data.
       if (uploadGeneration !== getUploadGeneration(getState())) {
         return false;
       }
 
       const { abortFunction, startUpload } = uploadBinaryProfileData();
       dispatch(uploadStarted(abortFunction));
 
-      if (uploadGeneration !== getUploadGeneration(getState())) {
-        // The upload could have been aborted while we were compressing the data.
-        return false;
-      }
-
```

Comment:
This was pretty useless with the same check a few lines before and no async code inbetween.